### DISCUSSION TITLE
Change certificate validity from command line on EJBCA

### DISCUSSION THREAD

**Discussion by twardy103** [Keyfactor/ejbca-ce#242] [Q&A] [answered]
Hi,

I am using EJBCA for maintaining certificates. Currently I have generated certprofile but I want to verify the scenario when encodedvalidity is set to another value (would be set from testcase point of view to see any alarms in the system). 
I am struggling with updating XML for certprofile because it is a hashmap written by XML decoder. I have checked EJBCA implementation where you are creating this xml file but it is a code which I want to avoid to use :).
Do you have any command line option to update the validity value? 
I mean - when I create an EJBCA pod, I want to login on it and run some ejbca.sh <parameters> command to set it.

Thanks in advance!

BR,
Aleksander

**Comment by primetomas** [comment, marked as answer]
There is not a simple CLI command to do this in Community, without editing the XML as you say. 

Enterprise have some additional automation tools that enables you to edit human readable YAML to update everything. 
https://doc.primekey.com/ejbca/ejbca-operations/ejbca-operations-guide/command-line-interfaces/configdump-tool
This is also available over REST APIs to make it easier. But these extra functions are not available in Community unfortunately.

Regards,
Tomas

**Reply by twardy103** [reply]
Thanks for your input!
I were able to write a code in Java to read certProfile, update it and save again to make it as an input to EJBCA, but later on I realized that I have changed wrong value... 
I have changed encoded validity in certprofile, but to be honest I wanted to change a parameter in "ejbca.sh ca init" command "-v <value>". 
BR,
Aleksander
